webapp/components/survey/NodeDefDetails/store/useNodeDefDetails.js

```diff
@@ -20,20 +20,33 @@ export const useNodeDefDetails = () => {
   const surveyCycleKey = useSurveyCycleKey()
 
   const [state, setState] = useState({})
+  const stateRef = useRef(state)
 
   const Actions = useActions({ setState })
 
   const editingFromDesigner = useIsDesignerNodeDefRoute()
 
   useEffect(() => {
     // Editing a nodeDef
-    if (nodeDefUuid) {
+    if (nodeDefUuid && !State.getNodeDef(state)) {
       const nodeDefSurvey = Survey.getNodeDefByUuid(nodeDefUuid)(survey)
       const validation = Survey.getNodeDefValidation(nodeDefSurvey)(survey)
       setState(State.create({ nodeDef: nodeDefSurvey, validation }))
     }
+
+    return () => {
+      // get local state from stateRef to get it up to date in this callback
```

Comment:
what does that mean?

Reply:
the local state that you get from the "state" variable is no up to date in this callback.
The only way I've found to get an updated state there was to put it in a "ref".

Reply:
of course. have you tried to pass the state to the hook input param?

Reply:
...the cancelEdits function should be called only on unmount, not every time the state changes...

Reply:
Do you necessarily need to pass the state to that action ?
Keeping a ref that is just a reference to the state looks a bit dirty to me